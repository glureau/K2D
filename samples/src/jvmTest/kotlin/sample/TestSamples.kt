package sample

import org.approvaltests.Approvals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import java.io.File

@RunWith(JUnit4::class)
class TestSamples {

    @Test
    fun `check all generated files`() {
        val resourcePath = "build/generated/ksp/metadata/commonMain/resources"
        val generatedDir = File(resourcePath)
        println(generatedDir.absolutePath)
        var allMarkdownGenerated = ""
        generatedDir.onEachFile { file ->
            println("Testing ${file.absolutePath}")
            allMarkdownGenerated += "File: " + file.absolutePath.substringAfter("$resourcePath/") + "\n" + file.readText() + "\n"
        }

        Approvals.verify(allMarkdownGenerated)

        // Should be... but a space is generated 50% of the time from the library ??
        /*
        val files = mutableListOf<File>()
        generatedDir.onEachFile {
            files += it
        }
        CombinationApprovals.verifyAllCombinations({ it.readText().trim() }, files.toTypedArray())
         */
    }

    private fun File.onEachFile(action: (file: File) -> Unit) {
        listFiles().forEach { child ->
            if (child.isDirectory) {
                child.onEachFile(action)
            } else {
                action(child)
            }
        }
    }
}