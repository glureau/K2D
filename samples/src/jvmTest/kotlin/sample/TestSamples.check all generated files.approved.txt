File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/Shapes23.md
```mermaid
classDiagram
  class Shape {
    <<interface>>
    +originPosition Position
    +computeSurface() Float
  }
  Shape ..* Position : has
  click Shape href "./sample/-shape/index.html"
  class Rectangle {
    <<interface>>
    +width Float
    +height Float
    +rotate(Float) 
  }
  Shape <|-- Rectangle : implements
  click Rectangle href "./sample/-rectangle/index.html"
  class Polygon {
    <<interface>>
    +howMuchSides() Int
  }
  Shape <|-- Polygon : implements
  click Polygon href "./sample/-polygon/index.html"
  class Square {
    <<class>>
    +sideSize Float
    +publicFun() 
  }
  Polygon <|-- Square : implements
  click Square href "./sample/-square/index.html"
  class Circle {
    <<data class>>
    +radius Float
  }
  Shape <|-- Circle : implements
  click Circle href "./sample/-circle/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/module.md
# Module samples

```mermaid
classDiagram
  class Position {
    <<data class>>
    +x Float
    +y Float
  }
  click Position href "./sample/-position/index.html"
  class Shape {
    <<interface>>
    +originPosition Position
    +computeSurface() Float
  }
  Shape ..* Position : has
  click Shape href "./sample/-shape/index.html"
  class Rectangle {
    <<interface>>
    +width Float
    +height Float
    +rotate(Float) 
  }
  Shape <|-- Rectangle : implements
  click Rectangle href "./sample/-rectangle/index.html"
  class Polygon {
    <<interface>>
    +howMuchSides() Int
  }
  Shape <|-- Polygon : implements
  click Polygon href "./sample/-polygon/index.html"
  class Square {
    <<class>>
    +sideSize Float
    +publicFun() 
  }
  Polygon <|-- Square : implements
  click Square href "./sample/-square/index.html"
  class Circle {
    <<data class>>
    +radius Float
  }
  Shape <|-- Circle : implements
  click Circle href "./sample/-circle/index.html"
  class Oval {
    <<data class>>
    +centerOne Position
    +centerTwo Position
    +distanceBetweenPoints Float
    +angle Float
    +radius Float
  }
  Shape <|-- Oval : implements
  Oval ..* Position : has
  Oval ..* Position : has
  click Oval href "./sample.another/-oval/index.html"
  class InnerOval {
    <<class>>
    +square Square
  }
  InnerOval ..* Square : has
  click InnerOval href "./sample.another/-inner-oval/index.html"
  class Builder {
    <<object>>
    +build() Oval
  }
  click Builder href "./sample.another/-builder/index.html"
  class TheTrap {
    <<class>>
    +foo Int
  }
  click TheTrap href "./sample.another/-the-trap/index.html"
  class BinaryIntCalculator {
    <<class>>
    +reg1 Int
    +reg2 Int
    +calculate((Int, () -> Int)-> Int) Int
  }
  click BinaryIntCalculator href "./sample.calculator/-binary-int-calculator/index.html"
  class GenericFoo~T:Number~ {
    <<interface>>
    +list List~T~
    +latestResult Result~Int~
    +lambda ⟮String⟯-> Double
    +add(T) 
    +addAll(List~T~) List~T~
    +computeResult() Result~Int~
  }
  click GenericFoo~T:Number~ href "./sample.calculator/-generic-foo/index.html"
  class Direction {
    <<enum>>
    [[ UP ]]
    [[ DOWN ]]
    [[ LEFT ]]
    [[ RIGHT ]]
  }
  click Direction href "./sample.enums/-direction/index.html"
  class DeviceKind {
    <<enum>>
    +userAgent String
    +index Int
    [[ Smartphone ]]
    [[ Computer ]]
    [[ Tablet ]]
    [[ Smartwatches ]]
    [[ Chromebook ]]
  }
  click DeviceKind href "./sample.enums/-device-kind/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Square.md
#### Square

##### Properties


| Name           | Type     | Comments |
|----------------|----------|----------|
| originPosition | Position |          |
| sideSize       | Float    |          |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |
| howMuchSides   | Int         |          |
| internalFun    | Unit        |          |
| privateFun     | Unit        |          |
| protectedFun   | Unit        |          |
| publicFun      | Unit        |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/calculator/table_BinaryIntCalculator.md
#### BinaryIntCalculator

##### Properties


| Name | Type | Comments |
|------|------|----------|
| reg1 | Int  |          |
| reg2 | Int  |          |

##### Functions


| Name      | Return Type | Comments |
|-----------|-------------|----------|
| calculate | Int         |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/calculator/table_GenericFoo.md
#### GenericFoo

##### Properties


| Name         | Type               | Comments |
|--------------|--------------------|----------|
| lambda       | (String) -> Double |          |
| latestResult | Result<Int>        |          |
| list         | List<T>            |          |

##### Functions


| Name          | Return Type | Comments |
|---------------|-------------|----------|
| add           | Unit        |          |
| addAll        | List<T>     |          |
| computeResult | Result<Int> |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/calculator/package.md
# Package sample.calculator

```mermaid
classDiagram
  class BinaryIntCalculator {
    <<class>>
    +reg1 Int
    +reg2 Int
    +calculate((Int, () -> Int)-> Int) Int
  }
  click BinaryIntCalculator href "./sample.calculator/-binary-int-calculator/index.html"
  class GenericFoo~T:Number~ {
    <<interface>>
    +list List~T~
    +latestResult Result~Int~
    +lambda ⟮String⟯-> Double
    +add(T) 
    +addAll(List~T~) List~T~
    +computeResult() Result~Int~
  }
  click GenericFoo~T:Number~ href "./sample.calculator/-generic-foo/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Polygon.md
#### Polygon

##### Properties


| Name           | Type     | Comments |
|----------------|----------|----------|
| originPosition | Position |          |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |
| howMuchSides   | Int         |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Position.md
#### Position

##### Properties


| Name | Type  | Comments |
|------|-------|----------|
| x    | Float |          |
| y    | Float |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/enums/table_Direction.md
#### Direction

| Entry | Comments |
|-------|----------|
| UP    |          |
| DOWN  |          |
| LEFT  |          |
| RIGHT |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/enums/table_DeviceKind.md
#### DeviceKind

##### Properties


| Name      | Type   | Comments |
|-----------|--------|----------|
| index     | Int    |          |
| userAgent | String |          |

| Entry        | Comments |
|--------------|----------|
| Smartphone   |          |
| Computer     |          |
| Tablet       |          |
| Smartwatches |          |
| Chromebook   |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/enums/package.md
# Package sample.enums

```mermaid
classDiagram
  class Direction {
    <<enum>>
    [[ UP ]]
    [[ DOWN ]]
    [[ LEFT ]]
    [[ RIGHT ]]
  }
  click Direction href "./sample.enums/-direction/index.html"
  class DeviceKind {
    <<enum>>
    +userAgent String
    +index Int
    [[ Smartphone ]]
    [[ Computer ]]
    [[ Tablet ]]
    [[ Smartwatches ]]
    [[ Chromebook ]]
  }
  click DeviceKind href "./sample.enums/-device-kind/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Circle.md
#### Circle

##### Properties


| Name           | Type     | Comments |
|----------------|----------|----------|
| originPosition | Position |          |
| radius         | Float    |          |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Rectangle.md
#### Rectangle

##### Properties


| Name           | Type     | Comments |
|----------------|----------|----------|
| height         | Float    |          |
| originPosition | Position |          |
| width          | Float    |          |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |
| rotate         | Unit        |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/table_Shape.md
#### Shape

##### Properties


| Name           | Type     | Comments |
|----------------|----------|----------|
| originPosition | Position |          |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/package.md
# Package sample

```mermaid
classDiagram
  class Position {
    <<data class>>
    +x Float
    +y Float
  }
  click Position href "./sample/-position/index.html"
  class Shape {
    <<interface>>
    +originPosition Position
    +computeSurface() Float
  }
  Shape ..* Position : has
  click Shape href "./sample/-shape/index.html"
  class Rectangle {
    <<interface>>
    +width Float
    +height Float
    +rotate(Float) 
  }
  Shape <|-- Rectangle : implements
  click Rectangle href "./sample/-rectangle/index.html"
  class Polygon {
    <<interface>>
    +howMuchSides() Int
  }
  Shape <|-- Polygon : implements
  click Polygon href "./sample/-polygon/index.html"
  class Square {
    <<class>>
    +sideSize Float
    +publicFun() 
  }
  Polygon <|-- Square : implements
  click Square href "./sample/-square/index.html"
  class Circle {
    <<data class>>
    +radius Float
  }
  Shape <|-- Circle : implements
  click Circle href "./sample/-circle/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/another/table_TheTrap.md
#### TheTrap

##### Properties


| Name | Type | Comments |
|------|------|----------|
| foo  | Int  |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/another/table_InnerOval.md
#### InnerOval

##### Properties


| Name   | Type   | Comments |
|--------|--------|----------|
| square | Square |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/another/package.md
# Package sample.another

```mermaid
classDiagram
  class Oval {
    <<data class>>
    +centerOne Position
    +centerTwo Position
    +distanceBetweenPoints Float
    +angle Float
    +radius Float
  }
  Shape <|-- Oval : implements
  Oval ..* Position : has
  Oval ..* Position : has
  click Oval href "./sample.another/-oval/index.html"
  class InnerOval {
    <<class>>
    +square Square
  }
  InnerOval ..* Square : has
  click InnerOval href "./sample.another/-inner-oval/index.html"
  class Builder {
    <<object>>
    +build() Oval
  }
  click Builder href "./sample.another/-builder/index.html"
  class TheTrap {
    <<class>>
    +foo Int
  }
  click TheTrap href "./sample.another/-the-trap/index.html"

```

File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/another/table_Builder.md
#### Builder

##### Functions


| Name  | Return Type | Comments |
|-------|-------------|----------|
| build | Oval        |          |


File: /Users/glureau/Projects/MermaidKsp/samples/build/generated/ksp/metadata/commonMain/resources/sample/another/table_Oval.md
#### Oval


 Please check [this link](https://en.wikipedia.org/wiki/Oval) for reference


##### Properties


| Name                  | Type     | Comments                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|-----------------------|----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| angle                 | Float    | In Euclidean geometry, an angle is the figure formed by two rays, called the sides of the angle, sharing a common endpoint, called the vertex of the angle.[1] Angles formed by two rays lie in the plane that contains the rays. Angles are also formed by the intersection of two planes. These are called dihedral angles. Two intersecting curves may also define an angle, which is the angle of the rays lying tangent to the respective curves at their point of intersection.<div></div><div></div> Angle is also used to designate the measure of an angle or of a rotation. This measure is the ratio of the length of a circular arc to its radius. In the case of a geometric angle, the arc is centered at the vertex and delimited by the sides. In the case of a rotation, the arc is centered at the center of the rotation and delimited by any other point and its image by the rotation. |
| centerOne             | Position |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| centerTwo             | Position |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| distanceBetweenPoints | Float    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
| originPosition        | Position | This is the original position use to originate the...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| radius                | Float    |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |

##### Functions


| Name           | Return Type | Comments |
|----------------|-------------|----------|
| computeSurface | Float       |          |


